File: examen2Final/PPregunta1/haskell/pregunta1Estructurado.hs
sumatresnumeros :: (Num a)=> a->a->a->a
sumatresnumeros a b c = a+b+c

restai :: (Num a)=> a->a
restai a =a-3

forloop 0 a b c = return ()
forloop n a b c =
    do 
    let s = sumatresnumeros a b c
    print(s)
    forloop (n-1)  b c s

main :: IO()
main = do 
    putStrLn "Introduzca un numero termino de la serie fibo3:"
    num <- getLine
    let nlim = read num :: Int
    let a =0
    let b =1
    let c =1
    print (a)
    print (b)
    print (c)
    forloop (restai nlim) a b c
    

File: examen2Final/PPregunta1/python/pregunta1Estructurado.py
# serie fibonaci estructurado
nlim=int(input("Introduzca el numero limite de la serie Fibonacci:"))
if (nlim<=1):
    print(0)    
elif (nlim==2):
    print(1)
elif (nlim==3):
    print(1)
else:
    res = [0,1,1]
    a,b,c=0,1,1
    for i in range(1,nlim-3):
        s= a+b+c
        print(s)
        res.append(s)
        a,b,c=b,c,s
    print(res)    

